src/R4Mvc.Tools/CodeGen/MethodBuilder.cs

```diff
@@ -15,7 +15,7 @@ public class MethodBuilder
         private IList<ParameterSyntax> _parameters = new List<ParameterSyntax>();
         private BlockSyntax _bodyBlock;
         private ExpressionSyntax _expressionBodySyntax;
-        private bool _useGeneratedAttributes = false, _useNonActionAttribute = false, _noBody = false;
+        private bool _useGeneratedAttributes = false, _useNonActionAndNonHandlerAttributes = false, _useNonHandlerAttribute = false, _noBody = false;
```

Comment:
Also, you've added the `_useNonHandlerAttribute`, but you never used it. I'd rather you have two separate options, reverting `_useNonActionAndNonHandlerAttributes` back to only control one attribute